lib/bolt/node/ssh.rb

```diff
@@ -253,16 +258,20 @@ def with_remote_task(task_file, stdin)
       end
     end
 
-    def run_command(command)
+    def run_command(command, options = {})
+      @run_as = options['_run_as'] || @conf_run_as
```

Comment:
Should we just pass this value through `execute` (which already takes an `options` hash) rather than clobbering the instance variable?

Reply:
It's all the extra things that use execute - with_remote_task, with_task_wrapper, with_remote_tempdir - that made that messy.

Reply:
Ah that's fair. Hopefully the transport API changes will simplify some of this.

Reply:
this will be moot soon as well since the instance var won't exist in transports